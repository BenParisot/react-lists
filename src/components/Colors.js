import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';


export default function Colors({ colors }) {

  const colorsList = colors.map(color => {
    return (
      <li key={color.name}>
        <Color name={color.name} hex={color.hex} rgb={color.rgb} />
      </li>

    );
  });
  return (
    <ul>
      {colorsList}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};
