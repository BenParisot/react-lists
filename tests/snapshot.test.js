import React from 'react';
import { shallow } from 'enzyme';
import Colors from '../src/components/Colors';

describe('Colors component', () => {
  it('generates a list of colors', () => {
    const colors = [
      { name: 'red', hex: '#ff0000', rgb: { red: 255, green: 0, blue: 0 } },
      { name: 'green', hex: '#00ff00', rgb: { red: 0, green: 255, blue: 0 } },
      { name: 'blue', hex: '#0000ff', rgb: { red: 0, green: 0, blue: 255 } }
    ];
    const wrapper = shallow(<Colors colors={colors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
