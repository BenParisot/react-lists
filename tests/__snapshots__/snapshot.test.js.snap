// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component generates a list of colors 1`] = `
<ul>
  <li
    key="red"
  >
    <Color
      hex="#ff0000"
      name="red"
      rgb={
        Object {
          "blue": 0,
          "green": 0,
          "red": 255,
        }
      }
    />
  </li>
  <li
    key="green"
  >
    <Color
      hex="#00ff00"
      name="green"
      rgb={
        Object {
          "blue": 0,
          "green": 255,
          "red": 0,
        }
      }
    />
  </li>
  <li
    key="blue"
  >
    <Color
      hex="#0000ff"
      name="blue"
      rgb={
        Object {
          "blue": 255,
          "green": 0,
          "red": 0,
        }
      }
    />
  </li>
</ul>
`;
